src/dsp_tools/xmllib/models/dsp_base_resources.py

```diff
@@ -50,11 +42,15 @@ def add_comments(self, comments: list[str]) -> AnnotationResource:
         return self
 
     def serialise(self) -> etree._Element:
+        self._check_for_and_convert_unexpected_input()
         res_ele = self._serialise_resource_element()
         res_ele.append(self._serialise_annotation_of())
         res_ele.append(_serialise_has_comment(self.comments, self.res_id))
         return res_ele
 
+    def _check_for_and_convert_unexpected_input(self) -> None:
+        self.comments = _transform_unexpected_input(self.comments, "comments", self.res_id)
```

Comment:
nitpick: I'm not sure it's worth extracting a one-liner here... I guess you did so for consistency, but to me it feels like over-kill.

Reply:
I can move it